### ISSUE TITLE
Elm syntax highlighting

### ISSUE THREAD

**Issue by malaire** [geany/geany#1981]
It would be nice to have support for Elm language: https://elm-lang.org/

**Comment by elextr** [comment]
Depends what you mean by "support".

If Elm is close enough to an existing language you can make a custom language, see the Geany manual.

If Elm needs its own highlighting you need to provide a [Scintilla](https://www.scintilla.org/) lexer (C++) for it and the code and data to integrate that into Geany.

If Elm needs its own symbol parsing you need to provide a [Universal Ctags](https://github.com/universal-ctags/ctags) parser and the code and data to integrate that into Geany.

**Comment by malaire** [comment]
I don't know enough of languages to know which of those three cases applies.

**Comment by virtualritz** [comment]
+1
From my experience with Scintilla and looking at Elm syntax I'd say it needs a specific lexer and Ctags parser.

**Comment by dptole** [comment]
@malaire @codebrainz I agree it would be very nice to have Elm syntax highlighting so I started thinking about how to create it based on Sublime's

Luckly Elm is very similar to Haskell so if you change your Elm filetype to Haskell (Document > Set Filetype > Programming Languages > Haskell) the result will be slightly satisfactory

In order to make this behavior automatic I had to edit the `/usr/share/geany/filetype_extensions.conf` and append `*.elm;` to the line `Haskell=*.hs;*.lhs;*.hs-boot;*.lhs-boot;`. Its important to end the line with the semicolon. After that I closed and reopened Geany. When I opened a new Elm file the Haskell highlight syntax was applied

Apparently one could fine tune the syntax highlighter for Elm by editing the file "/usr/share/geany/filetypes.haskell"

From here the more demanding individual could create an Elm synax highlighter by inferring the config files structure

**Comment by elextr** [comment]
Instead of breaking the haskell filetype, you could make a [custom filetype](https://www.geany.org/manual/current/index.html#custom-filetypes) using the haskell lexer and potentially tune a few more things.

**Comment by dptole** [comment]
Instead of _extending_ the haskell filetype...
